// @theledger/fabric-mock-stub/dist/utils/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var winston_1 = require("winston");
var util_1 = require("util");
/**
 * helper functions
 */
var Helpers = /** @class */ (function () {
    function Helpers() {
    }
    /**
     * Winston Logger with default level: debug
     *
     * @static
     * @param {string} name
     * @param {string} [level]
     * @returns {LoggerInstance}
     * @memberof Helpers
     */
    Helpers.getLoggerInstance = function (name, level) {
        return new winston_1.Logger({
            transports: [new winston_1.transports.Console({
                    level: level || 'debug',
                    prettyPrint: true,
                    handleExceptions: true,
                    json: false,
                    label: name,
                    colorize: true,
                })],
            exitOnError: false,
        });
    };
    ;
    /**
     * Check number of args
     * accepts array of numbers
     *
     * @static
     * @param {string[]} args
     * @param {(number | number[])} amount
     * @memberof Helpers
     */
    Helpers.checkArgs = function (args, amount) {
        if (util_1.isArray(amount)) {
            if (!amount.filter(function (a) {
                return args.length === a;
            }).length) {
                throw new Error("Incorrect number of arguments. Expecting " + amount);
            }
        }
        else {
            if (args.length != amount) {
                throw new Error("Incorrect number of arguments. Expecting " + amount);
            }
        }
    };
    return Helpers;
}());
exports.Helpers = Helpers;
